logistics.lua

```diff
@@ -67,4 +75,13 @@ function logistics.count_layouts()
     return num_layouts
 end
 
+function check_player_has_character(player)
+	if not player.character then
+		util.log("You can only use this when you are controlling your character")
+		return false
+	end
+	
+	return true
+end
```

Comment:
the GUI button doesn't even appear if the player has no character (https://github.com/francoiscampbell/Bluegistics/blob/5a9b8805342817f84baed492c6f542c5896e395e/control.lua#L6-L9), is this check necessary?

Reply:
For me, it only disappears when I get back my character. But good that you bring it up because now I found out when does that button disappear! :D

Here is a video showing how it works for me. I use Space Exploration's satellite mode as an example, but FARL's control train mode should do this too.
A thing to note that currently with FARL the player still seems to have a character for some reason while remote controlling a train, so loading and saving [works](https://user-images.githubusercontent.com/9215491/106356081-0b412a00-62fd-11eb-9d11-8b8f5773de6f.png) as expected, but ~~I have no idea where the items are actually stored~~ [the item storage](https://user-images.githubusercontent.com/9215491/106356133-89053580-62fd-11eb-9e6c-0aaae4fbba84.png) gets cleared when I exit control mode, and when I reenter the same train the storage gets empty, so I think that's just a bugin FARL.

https://user-images.githubusercontent.com/9215491/106355959-e5675580-62fb-11eb-8e5f-9d9bdebb4e63.mp4

Sorry for the low quality, I've set OBS earlier to use a lower bitrate and I don't remember the default one, but at least it's very small :D

Reply:
Ah, I didn't know some mods allow you to toggle between character and no character. I had only accounted for mode like Brave New World, where you don't have a character at all.

Let's keep this message in as a message to the player, but can you change `util.log` to `player.print`? `util.log` is only enabled in debug mode, so the user won't see those messages.

Also, can you convert the tabs to spaces in this PR?

Thanks

Reply:
Made the changes, sorry for not converting the tabs to spaces earlier, forgot about that one

Reply:
No worries, thanks for your contribution.